src/transactionContext.js

```diff
@@ -78,12 +79,14 @@ module.exports = function( sql, SqlContext ) {
 			},
 			error: {
 				_onEnter: function() {
-					this.err.message = "Seriate SqlContext Error. Failed on step '" + this.priorState + "'." + this.err.message;
-					this.transaction.rollback( function() {
-						this.emit( "error", this.err );
-					}.bind( this ) );
-					if ( this.connection.close ) {
-						this.connection.close();
+					var message = util.format( "TransactionContext Error. Failed on step \"%s\" with: \"%s\"", this.priorState, this.err.message );
+					log.error( message );
+					this.err.message = message;
+					this.emit( "error", this.err );
+					if ( this.transaction ) {
+						this.transaction.rollback( function() {
+							this.emit( "error", this.err );
```

Comment:
I think that you will want to check for the rollback error here too. This is where I was hitting it yesterday.